core/src/main/java/io/kestra/core/runners/pebble/functions/FileSizeFunction.java

```diff
@@ -0,0 +1,101 @@
+package io.kestra.core.runners.pebble.functions;
+
+import io.kestra.core.storages.FileAttributes;
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.utils.Slugify;
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import io.pebbletemplates.pebble.extension.Function;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+@Singleton
+public class FileSizeFunction implements Function {
```

Comment:
There is a lot of sharing logic between the **read** pebble function and this function, so the logic could be centralized into the new class but didn't want to overengineer it, since it's my first issue here.

Reply:
Yes, an abstract FileFunction could be a good idea.
But let's keep that for later, I'm OK to merge it as is.